source/Background/services/QuotesService.ts

```diff
@@ -0,0 +1,95 @@
+import {browser} from 'webextension-polyfill-ts';
+import {Message} from './types';
+import {QUOTES_SERVICE_MESSAGES} from './messages';
+
+export const QUOTES_FETCHING_INTERVAL = 600000;
+
+export const QUOTES_API_URLS = [
+  'https://www.dashcentral.org/api/v1/public',
+  'https://rates2.dashretail.org/rates?source=dashretail&symbol=DASHUSD',
+  'https://min-api.cryptocompare.com/data/generateAvg?fsym=DASH&tsym=USD&e=Binance,Kraken,Poloniex,Bitfinex',
+];
+
+class QuotesService {
+  private static instance: QuotesService;
+
+  static MESSAGES = QUOTES_SERVICE_MESSAGES;
+
+  private dashUsdRate = 0;
+
+  private initialized = false;
+
+  constructor() {
+    this.init();
+  }
+
+  public static getInstance(): QuotesService {
+    if (!QuotesService.instance) {
+      QuotesService.instance = new QuotesService();
+    }
+
+    return QuotesService.instance;
+  }
+
+  public getDashUsdRate(): number {
+    return this.dashUsdRate;
+  }
+
+  public init(): void {
+    if (!this.initialized) {
+      this.fetchQuotes().then(() => {
+        setInterval(() => this.fetchQuotes(), QUOTES_FETCHING_INTERVAL);
+        this.initialized = true;
+      });
+    }
+  }
+
+  async fetchQuotes(): Promise<void> {
+    for (let i = 0; i < QUOTES_API_URLS.length; i + 1) {
+      try {
+        const response = await fetch(QUOTES_API_URLS[i]);
+        if (response.ok) {
+          const data = await response.json();
+          const result = this.formatResponse(data, i);
+          this.dashUsdRate = result;
+          await browser.runtime.sendMessage({
+            type: QUOTES_SERVICE_MESSAGES.DASH_USD_RATE_UPDATED,
+            payload: result,
+          });
+          break;
+        }
+      } catch (e) {
+        i += 1;
```

Comment:
You don't need it there, `i` will be incremented in any case by for loop.

You need to somehow handle the case where all three API urls thrown an error